### ISSUE TITLE
  Polygon filling issue with orthographic projection persists

### ISSUE THREAD

**Issue by krischer** [SciTools/cartopy#579]
Hi all,

the issue mentioned in #492 unfortunately still exists. The transform has gotten much more stable but there are still quite some issues where the problem remains.

Some of them appear by running this snippets:

```python
import matplotlib.gridspec as gridspec
import matplotlib.pylab as plt
import cartopy
import cartopy.crs as ccrs

plt.figure(figsize=(15, 15))
gs = gridspec.GridSpec(2, 2)

coords = [
    (1.530, 2.034),
    (-89.719, 10.51),
    (139.040, 19.482),
    (-27.394, -5.054)
]

k = -1

for i in range(2):
    for j in range(2):
        k += 1
        if (k + 1) > len(coords):
            continue

        ax = plt.subplot(gs[i, j], projection=ccrs.Orthographic(
            coords[k][0],
            coords[k][1]))
        ax.set_global()

        scale = "110m"
        ax.add_feature(cartopy.feature.NaturalEarthFeature(
            "physical", "ocean", scale, edgecolor="None",
            facecolor="#CCCCCC"))
        ax.coastlines(resolution=scale, color="#222222")

plt.show()
```

Result with the latest master (the continents should be white):
![maps](https://cloud.githubusercontent.com/assets/800487/6366831/c975b466-bccb-11e4-9d02-3ae1b426bea9.png)

**Comment by krischer** [comment]
The script to generate the multi map plot mentioned in #492 (change the seed to generate maps with new transforms):

```python
import cartopy
import cartopy.crs as ccrs
import math
import matplotlib.pylab as plt
import matplotlib.gridspec as gridspec
import random


plt.figure(figsize=(30, 30))
gs = gridspec.GridSpec(10, 10)

random.seed(112233)

for i in range(10):
    for j in range(10):

        lng = random.random() * 360.0 - 180.0
        # Account for latitudinal variations
        lat = math.degrees(math.asin(2 * random.random() - 1))

        ax = plt.subplot(gs[i, j], projection=ccrs.Orthographic(lng, lat))
        ax.set_global()

        scale = "110m"
        ax.add_feature(cartopy.feature.NaturalEarthFeature(
            "physical", "ocean", scale, edgecolor="None", facecolor="#CCCCCC"))

        ax.coastlines(resolution=scale, color="#222222")
        plt.title("Lng: %.3f, Lat: %.3f" % (lng, lat))

plt.savefig("big_map.png")
```

**Comment by QuLogic** [comment]
Random plots? How about a [brute-force check](https://gist.github.com/QuLogic/62d585a6a67876aa84f0)? It only takes... 2-3 hours... Okay, maybe that's a bit of overkill. It produces 368 settings that fail (out of 64800).

**Comment by krischer** [comment]
Haha :-) Thats strictly speaking still only a small fraction of possible values.

Another issue here is that these values are very easily found if one just interactively zooms and pans a plot.

**Comment by pelson** [comment]
I think this is pretty well solved in v0.13.0. Though I didn't re-run your brute-force test @QuLogic.

Slightly improved in #635.

**Comment by QuLogic** [comment]
I'm not so sure it's fixed. I tried running #635, but instead of failures, I'm seeing hard errors `Unidentified problem with geometry, linestring being re-added. Please raise an issue.` and they seem to occur _more_ often than the failures did.
